### PR TITLE
fix: improve get container ip

### PR DESCRIPTION
**Pull request by nodece**
### Motivation

Get container ip sometimes empty,  I check code and found that we return the `inspect.NetworkSettings.IPAddress`. If this IP is empty, we should check the `inspect.NetworkSettings.Networks`, it shouldn't empty.

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -284,7 +284,17 @@ func (c *DockerContainer) ContainerIP(ctx context.Context) (string, error) {
 		return "", err
 	}
 
-	return inspect.NetworkSettings.IPAddress, nil
+	ip := inspect.NetworkSettings.IPAddress
+	if ip == "" {
+		for _, settings := range inspect.NetworkSettings.Networks {
+			if settings.IPAddress != "" {
+				ip = settings.IPAddress
```
Comment (mdelapenya) [inline docker.go:291]: What do you think about using a more descriptive variable name?
```suggestion
		for _, network := range inspect.NetworkSettings.Networks {
			if network.IPAddress != "" {
				ip = network.IPAddress
```

Comment (nodece) [reply docker.go:291]: Good point!

**Review by mdelapenya — COMMENTED** [review]
This looks good to me, although I'm missing a few tests covering the new behaviour.

Would you mind adding them? 🙏 

Thanks!

**Comment by nodece** [comment]
Looks like I should make a mock test, have you any ideas?

**Comment by baez90** [comment]
Have a look at the `TestContainerAttachedToNewNetwork` test in `docker_test.go` it already 'covers' what you are trying to avoid.
I think it has something to do with whether a container is in the default `bridge` network or in another non-default network. In the latter case your way of retrieving the IP is the only one working.

**Comment by mdelapenya** [comment]
I'd say this PR has been already solved in #491. Please let us know if we can close it, thanks!

**Pull request closed without merging**